php-frontend/src/main/java/org/sonar/php/tree/visitors/AssignmentExpressionVisitor.java

```diff
@@ -88,22 +77,25 @@ private void handleListAssignment(ListExpressionTree lhs, ExpressionTree rhs) {
   }
 
   private void assign(Tree lhs, ExpressionTree rhs) {
-    Symbol symbol = symbolTable.getSymbol(lhs);
+    Symbol symbol = getSymbol(lhs);
     if (symbol != null) {
-      assignedValuesBySymbol.computeIfAbsent(symbol, v -> new ArrayList<>()).add(rhs);
+      ((SymbolImpl) symbol).assignValue(rhs);
```

Comment:
Not important, but I think having to cast here is kind of odd. What about making `symbol` field on `VariableIdentifierTreeImpl` directly of type `SymbolImpl` as we are not working on interfaces here anyways.